java-frontend/src/main/java/org/sonar/java/PerformanceMeasure.java

```diff
@@ -162,14 +165,51 @@ public void stop() {
     }
 
     @Override
-    public void stopAndLog(@Nullable File workDir) {
+    public void stopAndLog(@Nullable File workDir, boolean appendMeasurementCost) {
+      if (appendMeasurementCost) {
+        setCurrent(measure);
+        appendMeasurementCost();
+      }
       stop();
       saveToFile(workDir, measure);
       if (LOG.isDebugEnabled()) {
         LOG.debug("Performance Measures:\n" + jsonFormat(toJson(measure)));
       }
     }
 
+    private static void appendMeasurementCost() {
+      int numberOfSamples = 99;
+      String[] sampleNames = IntStream.range(0, numberOfSamples).mapToObj(i -> "m" + i).toArray(String[]::new);
+      Duration totalDuration = start("#MeasurementCost_v1");
+      PerformanceMeasure measurementCost = currentMeasure;
+      Duration temporaryDuration = start("#temporary");
+      measurementCost.getOrCreateChild("nanoTime").add(median(IntStream.range(0, numberOfSamples).mapToLong(i -> {
+        long start = System.nanoTime();
+        return System.nanoTime() - start;
```

Comment:
No need for a variable, we could probably one line this one

Reply:
I think it's hard to understand with `-(System.nanoTime() - System.nanoTime())` because you have to understand the order of operands evaluation. So I keep it like this.